axelrod/strategies/axelrod_first.py

```diff
@@ -589,3 +589,136 @@ def strategy(self, opponent: Player) -> Action:
+        - and there are more than 10 rounds remaining in the match
+        - and the total number of defections differs from a 50-50 random sample
+          by at least 3.0 standard deviations.
+
+    A ‘fresh start’ is a sequence of two cooperations followed by an assumption
+    that the game has just started (everything is forgotten).
+
+    This strategy came 2nd in Axelrod’s original tournament.
+
+    Names:
+
+    - TidemanAndChieruzzi: [Axelrod1980]_
+    """
+
+    name = 'Tideman and Chieruzzi'
+    classifier = {
+        'memory_depth': float('inf'),
+        'stochastic': False,
+        'makes_use_of': {"game", "length"},
+        'long_run_time': False,
+        'inspects_source': False,
+        'manipulates_source': False,
+        'manipulates_state': False
+    }
+
+    def __init__(self) -> None:
+        super().__init__()
+        self.is_retaliating = False
+        self.retaliation_length = 0
+        self.retaliation_remaining = 0
+        self.current_score = 0
+        self.opponent_score = 0
+        self.last_fresh_start = 0
+        self.fresh_start = False
+
+    def _decrease_retaliation_counter(self):
+        """Lower the remaining owed retaliation count and flip to non-retaliate
+        if the count drops to zero."""
+        if self.is_retaliating:
+            self.retaliation_remaining -= 1
+            if self.retaliation_remaining == 0:
+                self.is_retaliating = False
+
+    def _fresh_start(self):
+        """Give the opponent a fresh start by forgetting the past"""
+        self.is_retaliating = False
+        self.retaliation_length = 0
+        self.retaliation_remaining = 0
+
+    def _score_last_round(self, opponent: Player):
+        """Updates the scores for each player."""
+        # Load the default game if not supplied by a tournament.
+        game = self.match_attributes["game"]
+        last_round = (self.history[-1], opponent.history[-1])
+        scores = game.score(last_round)
+        self.current_score += scores[0]
+        self.opponent_score += scores[1]
+
+    def strategy(self, opponent: Player) -> Action:
+        if not opponent.history:
+            return C
+
+        # Calculate the scores.
+        self._score_last_round(opponent)
+
+        # Check if we have recently given the strategy a fresh start.
+        if self.fresh_start:
+            self.fresh_start = False
+            return C  # Second cooperation
+
+        # Check conditions to give opponent a fresh start.
+        current_round = len(self.history) + 1
+        if self.last_fresh_start == 0:
+            valid_fresh_start = True
+        # There needs to be at least 20 rounds before the next fresh start
+        else:
+            valid_fresh_start = current_round - self.last_fresh_start >= 20
+
+        if valid_fresh_start:
+            valid_points = self.current_score - self.opponent_score >= 10
+            valid_rounds = self.match_attributes['length'] - current_round >= 10
```

Comment:
Wouldn't the behavior be different if I use -1? 
`valid_rounds = self.match_attributes['length'] - current_round >= 10` would be true if length was float('inf') and false if length was -1.